### PR TITLE
feat: introduce translated selects and file-input-field-with-list

### PR DESCRIPTION
**Pull request by HendrikThePendric**
Currently this one is still in draft, because it requires https://github.com/dhis2/ui-core/pull/787 to be merged in first, and then an updated version of `ui-core` here in `ui-widgets`.

This PR is quite opinionated, but has been discussed with @ismay and @Mohammer5 already when we were in Oslo. So, I think it is important that @varl has a look and considers the following points:
- We decided to only add documentation about the defaultProps that these components add and just refer back to ui-core for the rest.
- The way we've implemented this, it will not render anything to the DOM, but in React DevTools we will see an additional component. And bearing in mind that `ui-forms` will again add an additional component to the mix, we decided to use a displayName, so in DevTools it will be fairly obvious how the nesting works.

Also, I've opted for just including demo stories only. No e2e test and stories because these components don't have any internal logic.

One more point, a bit off topic... To deal with pluralisation, the `FileInputFieldWithList` doesn't use `defaultProps` but a helper function to get the correct strings. This works fine, but would make auto-documentation based on propTypes problematic. In this case we could even switch to using defaultProps and a fixed string (i.e. `No files(s) selected yet`), but this could possibly be a problem when re-implementing the way our docs are being generated. (see [this slack discussion](https://dhis2.slack.com/archives/CBM8LNEQM/p1581411245008400))

### CODE REVIEW

File: i18n/en.pot [hunk 1]
```diff
@@ -5,8 +5,23 @@ msgstr ""
 "Content-Type: text/plain; charset=utf-8\n"
 "Content-Transfer-Encoding: 8bit\n"
 "Plural-Forms: nplurals=2; plural=(n != 1)\n"
-"POT-Creation-Date: 2019-05-06T10:04:38.118Z\n"
-"PO-Revision-Date: 2019-05-06T10:04:38.118Z\n"
+"POT-Creation-Date: 2020-02-11T10:59:00.005Z\n"
+"PO-Revision-Date: 2020-02-11T10:59:00.005Z\n"
+
+msgid "No files selected yet"
+msgstr ""
```
Comment (cooper-joe) [inline i18n/en.pot]: I suggest removing 'yet', to make 'No files selected'.

File: i18n/en.pot [hunk 2]
```diff
@@ -5,8 +5,23 @@ msgstr ""
...
+
+msgid "No file selected yet"
```
Comment (cooper-joe) [inline i18n/en.pot]: Again, I suggest removing 'yet', to make 'No file selected'.

File: i18n/en.pot [hunk 3]
```diff
@@ -5,8 +5,23 @@ msgstr ""
...
+msgstr ""
+
+msgid "Upload files"
+msgstr ""
+
+msgid "Upload file"
```
Comment (cooper-joe) [inline i18n/en.pot]: I suggest, 'Upload a file'

File: i18n/en.pot [hunk 4]
```diff
@@ -28,3 +43,15 @@ msgstr ""
...
+msgid "Clear"
+msgstr ""
+
+msgid "Filter"
+msgstr ""
+
+msgid "Loading"
+msgstr ""
+
+msgid "No Match"
```
Comment (cooper-joe) [inline i18n/en.pot]: I suggest 'No match found'. (using sentence-case, not title-case).

**Comment by HendrikThePendric** [comment]
Thanks @cooper-joe. I've adjusted these texts in https://github.com/dhis2/ui-widgets/pull/309/commits/317a1f7dc985a1f6bb72ba51ce6b2ef3995d8580 because even if we end up moving these strings to ui-core (as suggested in https://github.com/dhis2/ui-core/issues/794), having the correct sentences in place will still be relevant.

**Comment by HendrikThePendric** [comment]
Closing this one, because it is most likely that this will be solved differently when we move to a mono-repo, see https://github.com/dhis2/ui-core/pull/822

**Pull request closed without merging**